src/compiler/compiler.ml

```diff
@@ -1743,32 +1767,48 @@ end = struct
       let b = aux t in
       (* Format.eprintf "Bool is %b@." b; *)
       b
+    in *)
+    let is_eigen_variable ~depth = function
+      | Const b -> 0 <= b && b < depth
+      | _ -> false
+    in
+    let is_unif_var = function
+      | AppUVar _ | UVar (_, _, _) | Discard | Arg (_, _)|AppArg (_, _) -> true
+      | Nil|Const _|Lam _|App (_, _, _)|Cons (_, _)|Builtin (_, _)|CData _ -> false
```

Comment:
`(uvar as X)` is not a catch all, but `(X as Y)` is (although not very useful).

Reply:
Should be fixed here: 39dc4742a7321dfd82480f0389f409e9fc69c104. 
Tests [test113.elpi](https://github.com/LPCIC/elpi/blob/39dc4742a7321dfd82480f0389f409e9fc69c104/tests/sources/functionality/test113.elpi) and [test114.elpi](https://github.com/LPCIC/elpi/blob/39dc4742a7321dfd82480f0389f409e9fc69c104/tests/sources/functionality/test114.elpi)